flatpak-builder: 1.0.14 -> 1.2.0

r-ryantm opened the pull request:
Automatic update generated by [nixpkgs-update](https://github.com/ryantm/nixpkgs-update) tools. This update was made based on information from https://github.com/flatpak/flatpak-builder/releases.


meta.description for flatpak-builder is: "Tool to build flatpaks from source"

meta.homepage for flatpak-builder is: "https://github.com/flatpak/flatpak-builder"

meta.changelog for flatpak-builder is: ""


###### Updates performed
- Version update

###### To inspect upstream changes

- [Release on GitHub](https://github.com/flatpak/flatpak-builder/releases/tag/1.2.0)

- [Compare changes on GitHub](https://github.com/flatpak/flatpak-builder/compare/1.0.14...1.2.0)

###### Impact

<details>
<summary>
<b>Checks done</b> (click to expand)
</summary>

---

- built on NixOS
- The tests defined in `passthru.tests`, if any, passed
- /nix/store/4c33h6chbn4zdycrva8z5vs8sf99nviy-flatpak-builder-1.2.0/bin/flatpak-builder passed the binary check.
- 1 of 1 passed binary check by having a zero exit code.
- 1 of 1 passed binary check by having the new version present in output.
- found 1.2.0 with grep in /nix/store/4c33h6chbn4zdycrva8z5vs8sf99nviy-flatpak-builder-1.2.0
- found 1.2.0 in filename of file in /nix/store/4c33h6chbn4zdycrva8z5vs8sf99nviy-flatpak-builder-1.2.0

---

</details>
<details>
<summary>
<b>Rebuild report</b> (if merged into master) (click to expand)
</summary>

```
3 total rebuild path(s)

1 package rebuild(s)

1 x86_64-linux rebuild(s)
1 i686-linux rebuild(s)
0 x86_64-darwin rebuild(s)
1 aarch64-linux rebuild(s)


First fifty rebuilds by attrpath
flatpak-builder
```

</details>

<details>
<summary>
<b>Instructions to test this update</b> (click to expand)
</summary>

---

Either **download from Cachix**:
```
nix-store -r /nix/store/4c33h6chbn4zdycrva8z5vs8sf99nviy-flatpak-builder-1.2.0 \
  --option binary-caches 'https://cache.nixos.org/ https://nix-community.cachix.org/' \
  --option trusted-public-keys '
  nix-community.cachix.org-1:mB9FSh9qf2dCimDSUo8Zy7bkq5CX+/rkCWyvRCYg3Fs=
  cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY=
  '
```
(The Cachix cache is only trusted for this store-path realization.)
For the Cachix download to work, your user must be in the `trusted-users` list or you can use `sudo` since root is effectively trusted.

Or, **build yourself**:
```
nix-build -A flatpak-builder https://github.com/r-ryantm/nixpkgs/archive/435cb7b1340f0d398fd755f52ba71bf42c9f7558.tar.gz
```

After you've downloaded or built it, look at the files and if there are any, run the binaries:
```
ls -la /nix/store/4c33h6chbn4zdycrva8z5vs8sf99nviy-flatpak-builder-1.2.0
ls -la /nix/store/4c33h6chbn4zdycrva8z5vs8sf99nviy-flatpak-builder-1.2.0/bin
```

---

</details>
<br/>



### Pre-merge build results

We have automatically built all packages that will get rebuilt due to
this change.

This gives evidence on whether the upgrade will break dependent packages.
Note sometimes packages show up as _failed to build_ independent of the
change, simply because they are already broken on the target branch.

Result of `nixpkgs-review` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>flatpak-builder</li>
  </ul>
</details>

---

###### Maintainer pings

cc @jtojnar for [testing](https://github.com/ryantm/nixpkgs-update/blob/master/doc/nixpkgs-maintainer-faq.md#r-ryantm-opened-a-pr-for-my-package-what-do-i-do).

jtojnar reviewed (changes requested):
Needs debugedit dependency..

jtojnar committed:
flatpak-builder: 1.0.14 → 1.2.0

- https://github.com/flatpak/flatpak-builder/releases/tag/1.1.1
- https://github.com/flatpak/flatpak-builder/releases/tag/1.1.2
- https://github.com/flatpak/flatpak-builder/releases/tag/1.2.0

Move patches just after src
Add debugedit dependency

jtojnar committed:
debugedit: unstable-2021-07-05 → 5.0

https://sourceware.org/git/?p=debugedit.git;a=shortlog;h=refs/tags/debugedit-5.0

jtojnar commented:
cc @deliciouslytyped for debugedit

deliciouslytyped commented:
Looks good to me superficially, I have nothing substantial to add. :+1:
